certification/internal/shell/has_prohibited_packages.go

```diff
@@ -2,31 +2,60 @@ package shell
 
 import (
 	"bufio"
-	"bytes"
-	"os/exec"
+	"strings"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
 type HasNoProhibitedPackagesCheck struct{}
 
 func (p *HasNoProhibitedPackagesCheck) Validate(image string) (bool, error) {
-	stdouterr, err := exec.Command("podman", "run", "-it", "--rm", "--entrypoint", "rpm", image, "-qa", "--queryformat", "%{NAME}\n").CombinedOutput()
+	line, err := p.getDataToValidate(image)
 	if err != nil {
 		log.Error("unable to get a list of all packages in the image")
 		return false, err
 	}
 
-	scanner := bufio.NewScanner(bytes.NewReader(stdouterr))
+	return p.validate(line)
+
+}
+
+func (p *HasNoProhibitedPackagesCheck) getDataToValidate(image string) (string, error) {
+	runOpts := cli.ImageRunOptions{
+		EntryPoint:     "rpm",
+		EntryPointArgs: []string{"-qa", "--queryformat", "%{NAME}\n"},
+		LogLevel:       "debug",
+		Image:          image,
+	}
+	runReport, err := podmanEngine.Run(runOpts)
+	if err != nil {
+		log.Error("unable to get a list of all packages in the image, error: ", err)
+		log.Debugf("Stdout: %s", runReport.Stdout)
+		log.Debugf("Stderr: %s", runReport.Stderr)
+		return "", err
+	}
+	return runReport.Stdout, nil
+}
+
+func (p *HasNoProhibitedPackagesCheck) validate(line string) (bool, error) {
+	scanner := bufio.NewScanner(strings.NewReader(line))
+	found := false
+	var pkgs []string
 	for scanner.Scan() {
 		for _, pkg := range prohibitedPackageList {
 			if pkg == scanner.Text() {
-				log.Warn("found a prohibited package in the container image: ", pkg)
-				return false, nil
+				found = true
+				pkgs = append(pkgs, pkg)
 			}
 		}
 	}
+	if found {
+		log.Warn("found at least one prohibited package in the container image")
+		log.Warn("found the following prohibited packages: ", pkgs)
+		return false, nil
```

Comment:
Thank you Emily for fixing this bug. 

Two more thing: for the sake of code clarity:
1) please replace variable `found` with `prohibitedPackageFound` 
2) remove this line and replace line 60 with `return !prohibitedPackageFound, nil`